sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/PropagatorFactory.java

```diff
@@ -27,9 +31,41 @@ static PropagatorFactory getInstance() {
   @Override
   public ContextPropagators create(
       PropagatorModel model, SpiHelper spiHelper, List<Closeable> closeables) {
-    List<String> compositeModel = requireNonNull(model.getComposite(), "composite propagator");
-    TextMapPropagator textMapPropagator =
-        TextMapPropagatorFactory.getInstance().create(compositeModel, spiHelper, closeables);
-    return ContextPropagators.create(textMapPropagator);
+    List<TextMapPropagatorModel> textMapPropagatorModels = model.getComposite();
+    Set<String> propagatorNames = new HashSet<>();
+    List<TextMapPropagator> textMapPropagators = new ArrayList<>();
+    if (textMapPropagatorModels != null && !textMapPropagatorModels.isEmpty()) {
+      textMapPropagatorModels.forEach(
+          textMapPropagatorModel -> {
+            TextMapPropagatorAndName propagatorAndName =
+                TextMapPropagatorFactory.getInstance()
+                    .create(textMapPropagatorModel, spiHelper, closeables);
+            textMapPropagators.add(propagatorAndName.getTextMapPropagator());
+            propagatorNames.add(propagatorAndName.getName());
+          });
+    }
+
+    String compositeList = model.getCompositeList();
+    if (compositeList != null) {
+      List<String> propagatorNamesList =
+          // Process string list same as we process OTEL_PROPAGATORS, trimming and filtering empty
+          // and 'none'
+          Arrays.asList(compositeList.split(",")).stream()
```

Comment:
Just a little simpler I think
```suggestion
          Stream.of(compositeList.split(","))
```